### PR TITLE
Presences of Nekopost

### PR DESCRIPTION
**Pull request by boon14872**
Presences of Nekopost

### CODE REVIEW

File: websites/N/Nekopost/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+    "$schema": "https://schemas.premid.app/metadata/1.3",
+    "service": "B",
+    "author": {
+      "name": "B",
+      "id": "424565046748643328"
+    },
```
Comment (SlowLife1661) [inline websites/N/Nekopost/dist/metadata.json:7]: It's not service name, and this username name is invalid

File: websites/N/Nekopost/dist/metadata.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
+    "$schema": "https://schemas.premid.app/metadata/1.3",
+    "service": "B",
+    "author": {
+      "name": "B",
+      "id": "424565046748643328"
+    },
+    "category": "other",
+    "logo": "https://www.nekopost.net/demo/nekopost_logo.png",
```
Comment (SlowLife1661) [inline websites/N/Nekopost/dist/metadata.json:9]: Find the new logo that is at least 512x512 and upload to imgur

File: websites/N/Nekopost/dist/metadata.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{
+    "$schema": "https://schemas.premid.app/metadata/1.3",
+    "service": "B",
+    "author": {
+      "name": "B",
+      "id": "424565046748643328"
+    },
+    "category": "other",
+    "logo": "https://www.nekopost.net/demo/nekopost_logo.png",
+    "thumbnail": "https://www.nekopost.net/demo/nekopost_logo.png",
```
Comment (SlowLife1661) [inline websites/N/Nekopost/dist/metadata.json:10]: Find the actual thumbnail, use the screenshot of home page if you can't find one, and upload to imgur

File: websites/N/Nekopost/dist/metadata.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    "version": "1.0.0",
+    "color": "#e7dca5",
+    "tags": [
+      "manga",
+      "thai"
+    ],
+    "url": "www.nekopost.net",
+    "description": {
+      "th": "Nekopost",
+      "en": "Nekopost"
```
Comment (SlowLife1661) [inline websites/N/Nekopost/dist/metadata.json:20]: Find the actual description

**Review by Slowlife01 — CHANGES_REQUESTED** [review]
Remove the `.js` file and fix EsLint issues ( check DeepScan )

Edit: Also proof of working

File: websites/N/Nekopost/dist/metadata.json [hunk 5]
```diff
@@ -0,0 +1,24 @@
+{
+    "$schema": "https://schemas.premid.app/metadata/1.3",
+    "service": "Nekopost",
+    "author": {
+      "name": "NPC_B",
+      "id": "424565046748643328"
+    },
+    "category": "other",
+    "logo": "https://imgur.com/uoFY9IP",
+    "thumbnail": "https://imgur.com/iZcmtnY",
```
Comment (SlowLife1661) [inline websites/N/Nekopost/dist/metadata.json:10]:
```suggestion
    "logo": "https://i.imgur.com/uoFY9IP.png",
    "thumbnail": "https://i.imgur.com/iZcmtnY.png",
```

File: websites/N/Nekopost/presence.ts [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+  clientId: "846071986902925312"
+}),
+browsingStamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "nekopost_logo"
+  };
+
+  presenceData.startTimestamp = browsingStamp;
```
Comment (SlowLife1661) [inline websites/N/Nekopost/presence.ts:11]:
```suggestion
  const presenceData: PresenceData = {
    largeImageKey: "nekopost_logo",
    startTimestamp: browsingStamp
  };
```

File: websites/N/Nekopost/presence.ts [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+
+  if (document.location.host === "www.nekopost.net") {
```
Comment (SlowLife1661) [inline websites/N/Nekopost/presence.ts:13]: Remove this statement

**Review by Slowlife01 — CHANGES_REQUESTED** [review]

**Comment by boon14872** [comment]
image

File: websites/N/Nekopost/presence.ts [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+        } else {
+          presenceData.details = `ONovel :${document.querySelector("head > title").textContent}!`;
+          presenceData.smallImageKey = "search";
+        }
+      } else {
+        presenceData.details = "กำลังเลือก Original Novel";
+        presenceData.smallImageKey = "search";
+      }
+    } else if (document.location.pathname.includes("/explore")) {
+      presenceData.details = "กำลังเลือก Project ";
```
Comment (SlowLife1661) [inline websites/N/Nekopost/presence.ts:73]:
```suggestion
      presenceData.details = "กำลังเลือก Project";
```

**Review by Slowlife01 — CHANGES_REQUESTED** [review]

File: websites/N/Nekopost/dist/metadata.json [hunk 9]
```diff
@@ -0,0 +1,24 @@
+{
+    "$schema": "https://schemas.premid.app/metadata/1.3",
```
Comment (SlowLife1661) [inline websites/N/Nekopost/dist/metadata.json:2]:
```suggestion
    "$schema": "https://schemas.premid.app/metadata/1.4",
```

**Review by Slowlife01 — APPROVED** [review]
Looks good to me

**Review by Bas950 — APPROVED** [review]
LGTM

**Pull request merged**